### ISSUE TITLE
Mocking methods with nullable type parameter hints

### ISSUE THREAD

**Issue by nijm** [mlively/Phake#250]
Phake::mock can't create stubs with nullable type hints. The following produces an error:

```
class PhakeTest_NullableTypes
{
    public function objectParameter(?PhakeTest_A $param) {

    }
}

$mock = Phake::mock('PhakeTest_NullableTypes');
```

`Declaration of PhakeTest_NullableTypes_PHAKE59a6c9a23e0651::objectParameter(PhakeTest_A $param) should be compatible with PhakeTest_NullableTypes::objectParameter(?PhakeTest_A $param)`

**Comment by mollierobbert** [comment]
I'd love this PR to be merged as well. Can you spare a minute to take a look @mlively?

**Comment by Thijs-Riezebeek** [comment]
This would be great and would allow us to keep using Phake for testing PHP 7.1 code.

**Comment by mlively** [comment]
Thank you! This has been merged and released as v3.0.1

**Comment by mollierobbert** [comment]
Awesome! Thanks Mike.
